ext/oj/object.c

```diff
@@ -536,7 +536,7 @@ static void hash_set_value(ParseInfo pi, Val kval, VALUE value) {
         }
         break;
     case T_HASH:
-        if (rb_cHash != rb_obj_class(parent->val)) {
+        if (T_HASH != rb_type(parent->val)) {
```

Comment:
This is already checked in the previous switch statement.
But I didn't understand why this line is checking it again...

Reply:
In some cases the Hash has been subclassed with the expectation that the set method will have a side effect. Checking the class explicitly catches that.

Reply:
I see.
Hmm, this PR breaks the operation with subclasses.
So, I will close this PR.

Thanks